### PR TITLE
Feature/improvements

### PR DESCRIPTION
**Pull request by SimonSchick**
#### PR checklist

- [x] Ran `npm run lint` and `npm run test`
- [x] Added tests if applicable

#### What changes did you make?

- Fixed up changelog a bit
- Added better readme
- Added url and email detection
- Added junk punctuation detection
- Added config file support

#### Is there anything you'd like reviewers to focus on?

Nah.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,2 +1,9 @@
 # 0.1.0
 - Initial release
+# 0.2.0
+- Magic
+# 0.3.0
+- Added sub tree ignore via `<!-- i18n-checker:disable -->`
+# 0.4.0
+- Ignore URLs, Emails and trim puntution
```
Comment (connor4312) [inline changelog.md]: punctuation

File: readme.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 ### ng-i18n-checker [![Build Status](https://travis-ci.org/WatchBeam/ng-i18n-checker.svg?branch=master)](https://travis-ci.org/WatchBeam/ng-i18n-checker)
 
+# What does it do?
+
+This utility helps you identifying missing translation strings in html files in a common ng>=2 applications.
+
+If you need to disable the check in a sub tree, use `<!-- i18n-checker:disable -->`.
```
Comment (connor4312) [inline readme.md]: Quick example here would probably be super useful

File: src/Validator.ts [hunk 3]
```diff
@@ -32,12 +34,16 @@ export class I18nValidator {
     public static readonly defaultTemplateMatcher = /\{\{.*?\}\}/g;
     public static readonly defaultAssumeTextCondition = /\w{2,}/;
     public static readonly defaultIgnoreComment = /^\s*i18n-checker(:| )disable\s*$/;
+    public static readonly urlRegEx = /\b\w+:\/\/\S+\b/ig;
+    public static readonly emailRegEx = /\b[-a-z\d~!$%^&*_=+}{'?]+(\.[-a-z\d~!$%^&*_=+}{'?]+)*@([a-z\d_]+(\.[-a-z\d_]+)*\.[a-zрф]{2,})\b/i;
```
Comment (connor4312) [inline src/Validator.ts]: How about having a regex ignore list and allowing people to specify custom additions or exclusions to that list in the CLI? May be easier than having bunches of different flags and properties.

File: readme.md (hunk 2 above)
Comment (SimonSchick) [inline readme.md]: Will in the next iteration.

**Pull request merged**